File: scripts/process-images.js
#!/usr/bin/env node

/**
 * Script para padronizar fundos das imagens para branco
 * Usa sharp para processamento de imagens
 */

const fs = require('fs');
const path = require('path');
const sharp = require('sharp');

// Verificar se sharp está instalado
try {
  require.resolve('sharp');
} catch (e) {
  console.log('❌ Sharp não está instalado. Instalando...');
  console.log('Execute: npm install sharp');
  process.exit(1);
}

const inputDir = 'public/images';
const outputDir = 'public/images_processed';

// Extensões de imagem suportadas
const imageExtensions = ['.jpg', '.jpeg', '.png', '.webp', '.bmp', '.tiff'];

/**
 * Processa uma imagem para fundo branco
 */
async function processImage(inputPath, outputPath) {
  try {
    // Criar diretório de saída se necessário
    const outputDir = path.dirname(outputPath);
    if (!fs.existsSync(outputDir)) {
      fs.mkdirSync(outputDir, { recursive: true });
    }

    // Processar a imagem
    await sharp(inputPath)
      .flatten({ background: { r: 255, g: 255, b: 255 } }) // Fundo branco
      .jpeg({ quality: 95, progressive: true })
      .toFile(outputPath);

    console.log(`✅ Processado: ${path.basename(inputPath)}`);
    return true;
  } catch (error) {
    console.log(`❌ Erro ao processar ${path.basename(inputPath)}: ${error.message}`);
    return false;
  }
}

/**
 * Processa recursivamente um diretório
 */
async function processDirectory(dir) {
  let processed = 0;
  let total = 0;

  const files = fs.readdirSync(dir, { withFileTypes: true });

  for (const file of files) {
    const fullPath = path.join(dir, file.name);

    if (file.isDirectory()) {
      // Processar subdiretório recursivamente
      const subResult = await processDirectory(fullPath);
      processed += subResult.processed;
      total += subResult.total;
    } else if (file.isFile()) {
      const ext = path.extname(file.name).toLowerCase();
      
      if (imageExtensions.includes(ext)) {
        total++;
        
        // Calcular caminho de saída
        const relativePath = path.relative(inputDir, fullPath);
        const outputPath = path.join(outputDir, relativePath);
        
        // Sempre salvar como .jpg
        const outputPathJpg = outputPath.replace(/\.[^/.]+$/, '.jpg');
        
        if (await processImage(fullPath, outputPathJpg)) {
          processed++;
        }
      }
    }
  }

  return { processed, total };
}

/**
 * Função principal
 */
async function main() {
  console.log('🖼️  Iniciando processamento de imagens...');
  console.log(`📁 Diretório de entrada: ${inputDir}`);
  console.log(`📁 Diretório de saída: ${outputDir}`);
  console.log('-'.repeat(50));

  // Verificar se o diretório de entrada existe
  if (!fs.existsSync(inputDir)) {
    console.log(`❌ Diretório não encontrado: ${inputDir}`);
    return;
  }

  // Processar todas as imagens
  const result = await processDirectory(inputDir);

  console.log('-'.repeat(50));
  console.log(`✅ Processamento concluído!`);
  console.log(`📊 Imagens processadas: ${result.processed}/${result.total}`);
  console.log(`📁 Resultado salvo em: ${outputDir}`);

  if (result.processed < result.total) {
    console.log(`⚠️  ${result.total - result.processed} imagens não puderam ser processadas`);
  }
}

// Executar se chamado diretamente
if (require.main === module) {
  main().catch(console.error);
}

module.exports = { processImage, processDirectory };

File: scripts/process_images.py
#!/usr/bin/env python3
"""
Script para padronizar fundos das imagens para branco
Processa todas as imagens do projeto e remove fundos, deixando fundo branco
"""

import os
import sys
from PIL import Image, ImageOps
import numpy as np
from pathlib import Path

def process_image_to_white_background(input_path, output_path):
    """
    Processa uma imagem removendo o fundo e colocando fundo branco
    """
    try:
        # Abrir a imagem
        img = Image.open(input_path)
        
        # Converter para RGBA se não for
        if img.mode != 'RGBA':
            img = img.convert('RGBA')
        
        # Criar uma nova imagem com fundo branco
        white_bg = Image.new('RGB', img.size, (255, 255, 255))
        
        # Se a imagem tem transparência, usar o canal alpha
        if img.mode == 'RGBA':
            # Criar uma máscara baseada no canal alpha
            alpha = img.split()[-1]
            
            # Criar uma imagem temporária para o fundo branco
            temp = Image.new('RGB', img.size, (255, 255, 255))
            
            # Colar a imagem original sobre o fundo branco
            temp.paste(img, mask=alpha)
            img = temp
        else:
            # Se não tem transparência, tentar remover fundo baseado em cor
            # Converter para numpy para processamento
            img_array = np.array(img)
            
            # Detectar fundo (assumindo que é a cor mais comum nos cantos)
            corners = [
                img_array[0, 0],      # canto superior esquerdo
                img_array[0, -1],      # canto superior direito
                img_array[-1, 0],      # canto inferior esquerdo
                img_array[-1, -1]      # canto inferior direito
            ]
            
            # Encontrar a cor mais comum nos cantos (provavelmente o fundo)
            from collections import Counter
            corner_colors = [tuple(corner) for corner in corners]
            most_common_color = Counter(corner_colors).most_common(1)[0][0]
            
            # Criar máscara para remover o fundo
            mask = np.all(img_array == most_common_color, axis=2)
            
            # Aplicar a máscara
            img_array[mask] = [255, 255, 255]  # Fundo branco
            
            # Converter de volta para PIL
            img = Image.fromarray(img_array)
        
        # Salvar a imagem processada
        img.save(output_path, 'JPEG', quality=95)
        print(f"✅ Processado: {input_path} -> {output_path}")
        return True
        
    except Exception as e:
        print(f"❌ Erro ao processar {input_path}: {str(e)}")
        return False

def process_directory(input_dir, output_dir):
    """
    Processa todas as imagens de um diretório
    """
    input_path = Path(input_dir)
    output_path = Path(output_dir)
    
    # Criar diretório de saída se não existir
    output_path.mkdir(parents=True, exist_ok=True)
    
    # Extensões de imagem suportadas
    image_extensions = {'.jpg', '.jpeg', '.png', '.webp', '.bmp', '.tiff'}
    
    processed_count = 0
    total_count = 0
    
    # Processar recursivamente
    for file_path in input_path.rglob('*'):
        if file_path.is_file() and file_path.suffix.lower() in image_extensions:
            total_count += 1
            
            # Calcular caminho relativo
            relative_path = file_path.relative_to(input_path)
            output_file = output_path / relative_path
            
            # Criar diretório pai se necessário
            output_file.parent.mkdir(parents=True, exist_ok=True)
            
            # Processar a imagem
            if process_image_to_white_background(file_path, output_file):
                processed_count += 1
    
    return processed_count, total_count

def main():
    """
    Função principal
    """
    # Diretórios
    input_dir = "public/images"
    output_dir = "public/images_processed"
    
    print("🖼️  Iniciando processamento de imagens...")
    print(f"📁 Diretório de entrada: {input_dir}")
    print(f"📁 Diretório de saída: {output_dir}")
    print("-" * 50)
    
    # Verificar se o diretório de entrada existe
    if not os.path.exists(input_dir):
        print(f"❌ Diretório de entrada não encontrado: {input_dir}")
        return
    
    # Processar todas as imagens
    processed, total = process_directory(input_dir, output_dir)
    
    print("-" * 50)
    print(f"✅ Processamento concluído!")
    print(f"📊 Imagens processadas: {processed}/{total}")
    print(f"📁 Imagens processadas salvas em: {output_dir}")
    
    if processed < total:
        print(f"⚠️  {total - processed} imagens não puderam ser processadas")

if __name__ == "__main__":
    main()
